src/as/renderers/webgl/WebGLShader.ts

```diff
@@ -0,0 +1,59 @@
+/// THREE.js.r105
+
+/**
+ * A lower level function to compile either a vertex or fragment shader.
+ *
+ * @param gl the WebGlRenderingContext which is used by the rendering engine
+ * @param type the enum of the shader type
+ * @param string the shader source code represented as a string
+ *  
+ * @example
+ * const gl = renderer.getContext();
+ *
+ * const glVertexShader = new THREE.WebGLShader( gl, gl.VERTEX_SHADER, vertexSourceCode );
+ * const glFragmentShader = new THREE.WebGLShader( gl, gl.FRAGMENT_SHADER, fragmentSourceCode );
+ *
+ * const program = gl.createProgram();
+ *
+ * gl.attachShader( program, glVertexShader );
+ * gl.attachShader( program, glFragmentShader );
+ *
+ * gl.linkProgram( program );
+ * 
+ * @see https://threejs.org/docs/#api/en/renderers/webgl/WebGLShader
+ * 
+ * @author Kara Rawson / https://github.com/ZoeDreams
+ */
+export class WebGlShader {
+	shader: WebGLShader
```

Comment:
The type of the property is the class that it is in. This would need to be something else.

But in hind sight, I think this may be one of the few cases where it shouldn't actually be a class, and just a plain function (called without `new`), like this:

```js
function webGLShader( gl, type, string ) {
	var shader = gl.createShader( type );
	gl.shaderSource( shader, string );
	gl.compileShader( shader );
	return shader;
}

const shader = webGLShader(gl, type, `...`)
```

Reply:
Ah, I checked where it is used, and in fact is called without `new`:

```js
	var glVertexShader = WebGLShader( gl, gl.VERTEX_SHADER, vertexGlsl, renderer.debug.checkShaderErrors );
	var glFragmentShader = WebGLShader( gl, gl.FRAGMENT_SHADER, fragmentGlsl, renderer.debug.checkShaderErrors );
```

It is unconventional to have the uppercase first letter (although some native JS builtins are like that)

Reply:
awwh,  yes, i agree with that. Ill update that now  thank you.

Reply:
doesn't type script still need a class reference for when other classes and functions use shaders? how does this work?

Reply:
I wasn't clear on the question. By any chance, are you thinking about it like in Java, where all things need to be from classes?

Reply:
No, having it as a function is a good call, also faster too.